src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnectionFactory.java

```diff
@@ -83,21 +101,49 @@ public ServiceBusJmsConnectionFactory(ConnectionStringBuilder connectionStringBu
      */
     public ServiceBusJmsConnectionFactory(String sasKeyName, String sasKey, String host, ServiceBusJmsConnectionFactorySettings settings) {
         this.settings = settings;
-        this.initialize(sasKeyName, sasKey, host, settings);
+        this.setKeyValue(sasKey);
+        this.keyName = sasKeyName;
+        this.initialize(host, settings);
     }
     
-    private void initialize(String sasKeyName, String sasKey, String host, ServiceBusJmsConnectionFactorySettings settings) {
-        if (sasKeyName == null || sasKeyName == null || host == null) {
-            throw new IllegalArgumentException("SAS Key, SAS KeyName and the host cannot be null for a ServiceBus connection factory.");
-        }
-        
+    /**
+     * Create a ServiceBusJmsConnectionFactory using shared access key and host name.
+     * @param tenantId Tenent id 
+     * @param clientId Client id.
+     * @param clientSecret Client secret
+     * @param host The host name of the ServiceBus namespace. Example: your-namespace-name.servicebus.windows.net
+     * @param settings The options used for this ConnectionFactory. Null can be used as default.
+     */
+    public ServiceBusJmsConnectionFactory(String tenantId, String clientId, String clientSecret, String host, ServiceBusJmsConnectionFactorySettings settings) {
+        this.settings = settings;
+        this.setIsToken(true);
+        this.setTenantId(tenantId);
+        this.setAadClientId(clientId);
+        this.setClientSecret(clientSecret);
+        this.initialize(host, settings);
+    }
+    
+    public ServiceBusJmsConnectionFactory(String token, String host, ServiceBusJmsConnectionFactorySettings settings, boolean isToken) {
+        this.settings = settings;
+        this.setIsToken(isToken);
+        this.setKeyValue (token);
+        this.userToken = true;
```

Comment:
>this.userToken = true; [](http://example.com/codeflow?start=8&length=22)

nit: Same here, some variables are set here versus some are set using setters, any reason?

Reply:
This was done for the user when they provide token but we no longer are supporting this so it will be removed